src/v4/http/types.ts

```diff
@@ -71,7 +72,7 @@ export interface V4InteractionWithCompleteRequest {
 
 export interface V4InteractionwithRequest {
   willRespondWith(
-    status: number,
+    status: number | Matcher<number>,
```

Comment:
Is this right? `Matcher<number>` isn't necessarily the same as `StatusCodeMatcher<T>`.

Also, if these aren't always http requests, then I'm not sure that `number` is right

Reply:
It can't be a non HTTP request, the type-state designed DSL/interface won't allow that.

Reply:
...but your other point is correct, thank you.